indexer/src/main/scala/importers/geonames/PolygonDataFlattener.scala

```diff
@@ -0,0 +1,198 @@
+
+      val geoidColumnNameToNamespaceMapping =
+        List(
+          "qs_gn_id" -> GeonamesNamespace,
+          "gn_id" -> GeonamesNamespace) ++
+          FeatureNamespace.values.map(namespace => (namespace.name -> namespace))
+      val geoidsFromFile = for {
+        (columnName, namespace) <- geoidColumnNameToNamespaceMapping
+        values <- feature.propMap.get(columnName).toList
+        value <- values.split(",")
+        if value.nonEmpty
+      } yield {
+        // if value already contains ':' it was human-readable to begin with
+        if (value.indexOf(':') != -1) {
+          value
+        } else {
+          "%s:%s".format(namespace.name, value)
+        }
+      }
+      val fidsFromFile = geoidsFromFile
+        .filterNot(_.isEmpty).map(_.replace(".0", ""))
+        .flatMap(i => StoredFeatureId.fromHumanReadableString(i, Some(defaultNamespace)))
+
+      val fids: Seq[StoredFeatureId] = if (fidsFromFileName.nonEmpty) {
+        fidsFromFileName
+      } else if (fidsFromFile.nonEmpty) {
+        fidsFromFile
+      } else {
+        Nil
+      }
+      
+      val (names, woeTypes) = if (fids.isEmpty && polygonMappingConfig.isDefined) {
+        (getFixedNames(polygonMappingConfig.get, feature), polygonMappingConfig.get.getWoeTypes)
+      } else {
+        (Nil, Nil)
+      }
+      
+      outputWriter.write("%s\t%s\t%s\t%s\t%s\t%s\n".format(
+        polygonId.toString,
+        source,
+        fids.map(_.longId.toString).mkString(","),
+        names.map(dn => "%s:%s".format(dn.lang, dn.name)).mkString("|"),
+        woeTypes.map(list => list.map(_.name).mkString(",")).mkString("|"),
+        geomBase64String
+      ))
+    }
+  }
+
+  def load(defaultNamespace: FeatureNamespace, f: File, outputWriter: Writer): Unit = {
+    val fparts = f.getName().split("\\.")
+    val extension = fparts.lastOption.getOrElse("")
+    val shapeFileExtensions = List("shx", "dbf", "prj", "xml", "cpg")
+
+    if ((extension == "json" || extension == "geojson") && !PolygonMappingConfig.isMappingFile(f)) {
+      processFeatureIterator(
+        defaultNamespace,
+        new GeoJsonIterator(f),
+        PolygonMappingConfig.getMappingForFile(f),
+        outputWriter
+      )
+    } else if (extension == "shp") {
+      processFeatureIterator(
+        defaultNamespace,
+        new ShapefileIterator(f),
+        PolygonMappingConfig.getMappingForFile(f),
+        outputWriter
+      )
+    } else if (shapeFileExtensions.has(extension)) {
+      // do nothing, shapefile aux file
+      Nil
+    }
+  }
+
+  def main(args: Array[String]): Unit = {
+    val defaultNameSpace = GeonamesNamespace
+    val polygonDirs = List(
+      new File("data/computed/polygons"),
+      new File("data/private/polygons"))
+    val polygonFiles = polygonDirs.flatMap(recursiveListFiles).sorted
+
+    val fileWriter = new FileWriter("data/private/flattenedPolygons.txt", false)
```

Comment:
it would be really cool to have literal file paths defined in one place and referred to symbolically elsewhere